### DISCUSSION TITLE
[Request] How do I put values on top of the bar chart ?

### DISCUSSION THREAD

**Discussion by dickensas** [d3/d3#3905] [Q&A] [answered]
This is not an issue, this is only a help/request needed

When I create a bar chart with label I always get the label inside the bar

How do I put data labels on top of each bar as shown below, without writing svg DOM code

![image](https://github.com/user-attachments/assets/f6036f21-4524-4a27-9da7-909e82604b6a)

My question is, is the default parameter facility available in D3 ?
Or should I write SVG based DOM manipulation code

**Comment by Fil** [comment, marked as answer]
Yes with D3 you necessarily have to write some code.
